### PR TITLE
Feature: artifact download API

### PR DESCRIPTION
**Pull request by artemii-yanushevskyi**
[Discussion](https://nevercode.slack.com/archives/CCGU5M8P3/p1675340231206539?thread_ts=1675265969.090359&cid=CCGU5M8P3)

### CODE REVIEW

File: content/rest-api/artifacts.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+#### Parameters
+
+| **Name**    | **Type**  | **Description**                          |
+| ----------- | --------- | ---------------------------------------- |
+| `expiresAt` | `integer` | Optional. URL expiration UNIX timestamp. |
+
+The response contains public artifact download URL under `url` key.
+
+This URL can be shared with trusted people who don't have Codemagic account to download artifacts directly.
```
Comment (nevercode-kevin) [inline content/rest-api/artifacts.md:25]:
```suggestion
This URL can be shared with trusted people who don't have a Codemagic account to download artifacts directly.
```

File: content/rest-api/artifacts.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+title: Artifacts API
+description: Authenticated and public access to build artifacts
+weight: 3
+---
+
+## Authenticated download URL
+
+`GET /artifacts/:uuid1/:uuid2/:artifactName`
+
+This URL can be obtained using [Builds API](/rest-api/builds) or copied directly from Codemagic UI.
```
Comment (nevercode-kevin) [inline content/rest-api/artifacts.md:11]:
```suggestion
This URL can be obtained using the [Builds API](/rest-api/builds) or copied directly from the Codemagic UI.
```

File: content/rest-api/artifacts.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+`POST /artifacts/:uuid1/:uuid2/:artifactName/public-url`
+
+#### Parameters
+
+| **Name**    | **Type**  | **Description**                          |
+| ----------- | --------- | ---------------------------------------- |
+| `expiresAt` | `integer` | Optional. URL expiration UNIX timestamp. |
+
+The response contains public artifact download URL under `url` key.
```
Comment (nevercode-kevin) [inline content/rest-api/artifacts.md:23]:
```suggestion
The response contains the public artifact download URL under `url` key.
```

File: content/rest-api/artifacts.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+This URL can be shared with trusted people who don't have Codemagic account to download artifacts directly.
+
+{{<notebox>}}
+**Note:** URLs generated without providing `expiresAt` UNIX timestamp expire in 24 hours.
```
Comment (nevercode-kevin) [inline content/rest-api/artifacts.md:28]:
```suggestion
**Note:** URLs generated without providing an `expiresAt` UNIX timestamp expire in 24 hours.
```

**Review by nevercode-kevin — CHANGES_REQUESTED** [review]

File: content/rest-api/artifacts.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+`POST /artifacts/:uuid1/:uuid2/:artifactName/public-url`
+
+#### Parameters
+
+| **Name**    | **Type**  | **Description**                          |
+| ----------- | --------- | ---------------------------------------- |
+| `expiresAt` | `integer` | Optional. URL expiration UNIX timestamp. |
+
+The response contains the public artifact download URL under `url` key.
```
Comment (helinanever) [inline content/rest-api/artifacts.md:23]:
```suggestion
The response contains the public artifact download URL under the `url` key.
```

File: content/rest-api/artifacts.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+`GET /artifacts/:uuid1/:uuid2/:artifactName`
+
+This URL can be obtained using the [Builds API](/rest-api/builds) or copied directly from the Codemagic UI.
+
+## Retrieve public download URL
+
+`POST /artifacts/:uuid1/:uuid2/:artifactName/public-url`
+
+#### Parameters
```
Comment (helinanever) [inline content/rest-api/artifacts.md:17]: This should be ###

File: content/rest-api/artifacts.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+The response contains the public artifact download URL under `url` key.
+
+This URL can be shared with trusted people who don't have a Codemagic account to download artifacts directly.
+
+{{<notebox>}}
+**Note:** URLs generated without providing an `expiresAt` UNIX timestamp expire in 24 hours.
+{{</notebox>}}
+
+#### Example
```
Comment (helinanever) [inline content/rest-api/artifacts.md:31]: ###

File: content/rest-api/artifacts.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+{{< highlight bash "style=paraiso-dark">}}
+curl https://api.codemagic.io/artifacts/12c0b167-b33c-4761-88c4-f3ac2478aeee/bc35a099-3ab9-41b2-b42f-f45ef59b5bac/Kefir.ipa/public-url \
+  -H "Content-Type: application/json" \
+  -H "X-Auth-Token: <API Token>" \
+  -d '{"expiresAt": 1675419345}'
+{{< /highlight >}}
+
+#### Response
```
Comment (helinanever) [inline content/rest-api/artifacts.md:40]: ###

File: content/rest-api/artifacts.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+title: Artifacts API
+description: Authenticated and public access to build artifacts
+weight: 3
+---
+
+## Authenticated download URL
+
+`GET /artifacts/:uuid1/:uuid2/:artifactName`
+
+This URL can be obtained using the [Builds API](/rest-api/builds) or copied directly from the Codemagic UI.
```
Comment (helinanever) [inline content/rest-api/artifacts.md:11]: Should we have an example for this as well?

**Review by helinanever — CHANGES_REQUESTED** [review]

File: content/rest-api/artifacts.md (hunk 9 above)
Comment (artemii-yanushevskyi) [reply content/rest-api/artifacts.md:11]: It's just a URL to copy-paste in browser to download artifact right away, no `curl` requests required

File: content/rest-api/artifacts.md (hunk 6 above)
Comment (artemii-yanushevskyi) [reply content/rest-api/artifacts.md:17]: It's like that everywhere on the [/rest-api/builds.md](https://github.com/codemagic-ci-cd/codemagic-docs/blob/master/content/rest-api/builds.md?plain=1#L17) page

**Review by helinanever — APPROVED** [review]
LGTM

**Pull request merged**